src/functions.php

```diff
@@ -503,6 +503,63 @@ function boolean_value($input)
     return null;
 }
 
+/**
+ * Parses ini sections with subsections (i.e. the service section)
+ *
+ * @param $filename
+ * @param $filename
+ * @return array
+ */
+function parse_ini_section_with_subsections($filename, $section_name) {
+    $config = [];
+    $stream = fopen($filename, 'r');
+
+    if (!$stream) {
+        return $config;
+    }
+
+    $current_subsection = '';
+
+    while (!feof($stream)) {
+        $line = trim(fgets($stream));
```

Comment:
should we condense these few lines into another function called read_ini_line?  Just to avoid the duplication of the trim, the fgets, and the empty/comment check